packages/web/src/components/createCell.tsx

```diff
@@ -11,16 +11,22 @@ import { useCellCacheContext } from './CellCacheContext'
  */
 import { useQuery } from './GraphQLHooksProvider'
 
+/**
+ * GraphQL Variables required to be included in the Cell props.
+ */
+type CellPropsVariables<Cell, GQLVariables> = Cell extends {
```

Comment:
Not super clear to me what this ternary is doing!

I haven't looked at this in a long time, so a little bit of an explanation (maybe with the type tests) would be really helpful!

Reply:
`CellPropsVariables` represents the type of the variables required to be included in the Cell props.

if `beforeQuery` exists and it returns `variables`, all the variables should be defined in the return value. Thus any partial objects of `variables` do not have to be included in Cell props. For that case, this type becomes empty.
Otherwise, all the `variables` must be in included in the Cell's props, thus this type becomes `GQLVariables`.